### PR TITLE
Migrate Dropdown to ScrollConfiguration2

### PR DESCRIPTION
**Pull request by abarth**
Also use the word "dropdown" consistently over "drop down".

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/lib/src/material/dropdown.dart [hunk 1]
```diff
@@ -79,23 +79,19 @@ class _DropdownMenuPainter extends CustomPainter {
...
 
   @override
-  ExtentScrollBehavior createScrollBehavior() => new OverscrollWhenScrollableBehavior(platform: platform);
+  TargetPlatform getPlatform(BuildContext context) => platform;
 
   @override
-  Widget wrapScrollWidget(BuildContext context, Widget scrollWidget) {
-    return new ClampOverscrolls(edge: ScrollableEdge.both, child: scrollWidget);
-  }
+  Widget buildViewportChrome(BuildContext context, Widget child, AxisDirection axisDirection) => child;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/dropdown.dart]: no scroll bar?

File: packages/flutter/lib/src/material/dropdown.dart [hunk 2]
```diff
@@ -79,23 +79,19 @@ class _DropdownMenuPainter extends CustomPainter {
...
+  const _DropdownScrollBehavior(this.platform);
 
-  @override
-  TargetPlatform get platform => _platform;
-  final TargetPlatform _platform;
+  final TargetPlatform platform;
 
   @override
-  ExtentScrollBehavior createScrollBehavior() => new OverscrollWhenScrollableBehavior(platform: platform);
+  TargetPlatform getPlatform(BuildContext context) => platform;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/dropdown.dart]: shouldn't this come from the theme?

File: packages/flutter/lib/src/material/dropdown.dart [hunk 3]
```diff
@@ -79,23 +79,19 @@ class _DropdownMenuPainter extends CustomPainter {
...
 
   @override
-  Widget wrapScrollWidget(BuildContext context, Widget scrollWidget) {
-    return new ClampOverscrolls(edge: ScrollableEdge.both, child: scrollWidget);
-  }
+  Widget buildViewportChrome(BuildContext context, Widget child, AxisDirection axisDirection) => child;
 
   @override
-  bool updateShouldNotify(ScrollConfigurationDelegate old) => platform != old.platform;
+  ScrollPhysics getScrollPhysics(BuildContext context) => const ClampingScrollPhysics();
```
Comment (Hixie) [inline packages/flutter/lib/src/material/dropdown.dart]: iOS dropdowns bounce, no?

File: packages/flutter/lib/src/material/dropdown.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/dropdown.dart]: see comment below

File: packages/flutter/lib/src/material/dropdown.dart [hunk 4]
```diff
@@ -182,8 +178,8 @@ class _DropdownMenuState<T> extends State<_DropdownMenu<T>> {
         child: new Material(
           type: MaterialType.transparency,
           textStyle: route.style,
-          child: new ScrollConfiguration(
-            delegate: new _DropdownScrollConfigurationDelegate(Theme.of(context).platform),
+          child: new ScrollConfiguration2(
+            behavior: new _DropdownScrollBehavior(Theme.of(context).platform),
```
Comment (Hixie) [inline packages/flutter/lib/src/material/dropdown.dart]: is it more correct to get the Theme here, or to use the context given to the scroll behavior?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/dropdown.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/material/dropdown.dart]: It's possible we should add a scroll bar, but the existing code didn't have one either.

File: packages/flutter/lib/src/material/dropdown.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/material/dropdown.dart]: Yes, that's probably a better approach.

File: packages/flutter/lib/src/material/dropdown.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter/lib/src/material/dropdown.dart]: The old code as explicit about not doing it.  I couldn't find any examples one way or another...

**Pull request merged**